### PR TITLE
[Windows] Notify changes in CollectionView Layouts 

### LINKED ISSUE #4112: [Windows] CollectionView ItemSpacing is not working
### Description

CollectionView ItemSpacing is not working
image

### Steps to Reproduce

1. Download [CollectionViewDemos.zip](https://github.com/dotnet/maui/files/7861241/CollectionViewDemos.zip)
2. Launch the App and navigate to any spacing sample.

### Version with bug

Preview 12 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10 1909

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #8387: CollectionView with GridItemsLayout: Issues when changing Span
### Description

I have a CollectionView with a GridItemsLayout. Depending on the available size I want to show more (landscape view) or less (portrait view) items on a row.

In the SizeChanged event I compute and set the right Span.

* **On iOS**: crash with NullReferenceException: CollectionView is null in GridViewLayout.HandlePropertyChanged().
* **On Windows**: Number of items on the row is NOT changed.
* **On Android**: When I add a short delay before changing Span it seems to work correct.

### Steps to Reproduce

See example project [MauiGridItemsLayoutChangeSpanIssue](https://github.com/Forestbrook/MauiGridItemsLayoutChangeSpanIssue)

In the AppShell Flyout you can select two views:
* CollectionView with workaround and
* CollectionView with bug

Rotate the phone/tablet to see the number of items on a row change.

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Windows

### Affected platform versions

iOS 15.5, Windows 10.0.19044

### Did you find any workaround?

For iOS I found a workaround by adding a check for CollectionView==null in GridViewLayout.HandlePropertyChanged(). See ```GridViewLayoutExt``` in the example project. This works, because HandlePropertyChanged() for the Span property is called again when CollectionView is not null anymore, which is kinda weird.

For Windows I did not look for a workaround yet.

### Relevant log output

_No response_

### LINKED ISSUE #10144: CollectionView items layout is not changed when GridItemsLayout.Span is changed on iOS and Windows
### Description

CollectionView items layout is not changed when GridItemsLayout.Span is changed on iOS and Windows - columns number remains the same.
Works fine on Android.

Additional problem: trying to change the span without Dispatcher causes crash on iOS (!)

Reproduced on:
Visual Studio 17.3.4
Windows 11 desktop
iPhone device with iOS 15.6.1

Attached: 
MauiApp1 source code that demonstrates the bug
[MauiApp1.zip](https://github.com/dotnet/maui/files/9574398/MauiApp1.zip)

### Steps to Reproduce

Bug 1
1. Open MauiApp1 application
2. Click the button to change layout span
3. Items layout is not changed on iOS and Windows

Bug 2
1. Go to OnButtonClicked implementation, eliminate dispatching and try to change span directly.
2. Recompile
3. Open MauiApp1 on iOS
4.  Click the button to change layout span
5. Application is crashing

[MauiApp1.zip](https://github.com/dotnet/maui/files/9574435/MauiApp1.zip)

### Link to public reproduction project repository

https://github.com/dotnet/maui/files/9574398/MauiApp1.zip

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Windows

### Affected platform versions

iOS 15.6.1, Windows SDK 10.0.22000.194

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Notify changes in CollectionView Layouts (Span, Spacing etc). 

Issue fixed on Windows (changing Spans).
fix-10144

To validate the changes, launch the .NET MAUI Gallery and navigate to the CollectionView section. Select the "Vertical Grid" sample and update the Span property value.
test-span

### Issues Fixed

FIxes #4112
Fixes #8387 
Fixes #10144

### CODE REVIEW

File: src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs [hunk 1]
```diff
@@ -17,6 +19,32 @@ public partial class StructuredItemsViewHandler<TItemsView> : ItemsViewHandler<T
 
 		protected override IItemsLayout Layout { get => ItemsView?.ItemsLayout; }
 
+		protected override void ConnectHandler(ListViewBase platformView)
+		{
+			base.ConnectHandler(platformView);
+
+			if (Layout != null)
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs:26]: Move to is not over is.
```suggestion
			if (Layout is not null)
```

File: src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs [hunk 2]
```diff
@@ -17,6 +19,32 @@ public partial class StructuredItemsViewHandler<TItemsView> : ItemsViewHandler<T
...
+				Layout.PropertyChanged += LayoutPropertyChanged;
+		}
+
+		protected override void DisconnectHandler(ListViewBase platformView)
+		{
+			base.DisconnectHandler(platformView);
+
+			if (Layout != null)
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs:34]: Same

File: src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs [hunk 3]
```diff
@@ -229,5 +257,33 @@ static WStyle GetHorizontalItemContainerStyle(LinearItemsLayout layout)
 
 			return style;
 		}
+
+		void UpdateItemsLayoutSpan()
+		{
+			if (ListViewBase is FormsGridView formsGridView)
+			{
+				formsGridView.Span = ((GridItemsLayout)Layout).Span;
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs:265]: Why not using a double is and remove the dangerous cast?
```csharp
if (ListViewBase is FormsGridView formsGridView && Layout is GridItemsLayout girdLayout)
```

File: src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs [hunk 4]
```diff
@@ -17,6 +19,32 @@ public partial class StructuredItemsViewHandler<TItemsView> : ItemsViewHandler<T
...
+		}
+
+		void LayoutPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == GridItemsLayout.SpanProperty.PropertyName)
+				UpdateItemsLayoutSpan();
+			else if (e.PropertyName == GridItemsLayout.HorizontalItemSpacingProperty.PropertyName || e.PropertyName == GridItemsLayout.VerticalItemSpacingProperty.PropertyName)
+				UpdateItemsLayoutItemSpacing();
+			else if (e.PropertyName == LinearItemsLayout.ItemSpacingProperty.PropertyName)
+				UpdateItemsLayoutItemSpacing();
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs:45]: Would look a lot nicer with a switch expression and the removal of the else clauses.

Comment (jsuarezruiz) [reply src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs:45]: Agree with that. In this case, just continued following the if-else approach used so far in Xamarin. https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Renderers/ButtonRenderer.cs#L140

**Review by mandel-macaque — APPROVED** [review]
Looks good. Some comments about using a switch expression, but that has nothing to do with how the code will behave and is just a note.

**Review by hartez — APPROVED** [review]

**Pull request merged**